kyo-prelude/shared/src/main/scala/kyo2/Memo.scala

```diff
@@ -0,0 +1,32 @@
+package kyo2
+
+import kyo.Tag
+import kyo2.kernel.*
+
+opaque type Memo = Var[Memo.Cache]
+
+object Memo:
+
+    private[kyo2] case class Cache(map: Map[(Any, Any), Any])
```

Comment:
Can this be an AnyVal? Or will it be boxed anyways?

Reply:
Yeah, it's used as a generic in `Var` so it'd box when handling the effect